### PR TITLE
[FLINK-19232][python] support iterating MapState and MapView

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request supports iterating MapState and MapView*

## Brief change log

  - *Support iterating MapState and MapView*

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-python/pyflink/fn_execution/state_impl.py [hunk 1]
```diff
@@ -301,6 +380,124 @@ def _convert_to_cache_key(state_key):
...
+        """
+        if self._last_key is None:
+            raise Exception("You need to call the '__next__' method before calling "
+                            "this method.")
+        self._check_modification()
+        # Bypass the 'remove' method of the map state to avoid triggering the commit of the write
+        # cache.
+        if self._internal_map_state._cleared:
+            del self._internal_map_state._write_cache[self._last_key]
+            self._mod_count += 1
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/state_impl.py:451]: Could refactor a bit to remove the duplicate code

File: flink-python/pyflink/fn_execution/state_impl.py [hunk 2]
```diff
@@ -395,6 +619,17 @@ def commit(self):
         self._write_cache.clear()
         self._cleared = False
 
+    def _iterate_next_batch(self, iterate_type, iterator_token):
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/state_impl.py:622]:
```suggestion
    def next_batch(self, iterate_type, iterator_token):
```

File: docs/_includes/generated/python_configuration.html [hunk 3]
```diff
@@ -57,13 +57,19 @@
             <td>If set, the Python worker will configure itself to use the managed memory budget of the task slot. Otherwise, it will use the Off-Heap Memory of the task slot. In this case, users should set the Task Off-Heap Memory using the configuration key taskmanager.memory.task.off-heap.size.</td>
         </tr>
         <tr>
-            <td><h5>python.map-state.read.cache.size</h5></td>
+            <td><h5>python.map-state.iterate-cache-size</h5></td>
```
Comment (dianfu) [inline docs/_includes/generated/python_configuration.html:60]: python.map-state.iterate-req-batch-size

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your comments! I have updated this PR.

**Review by dianfu — APPROVED** [review]

**Pull request closed without merging**